build.gradle

```diff
@@ -39,6 +39,20 @@ ext {
     generatedMetamodels = "build/generated/sources/annotationProcessor/java/main/org/candlepin/subscriptions"
 }
 
+spotless {
+    java {
+        target project.fileTree(project.rootDir) {
+            include '**/*.java'
+            exclude '**/build/**'
+        }
+        //allows build task to be successful, even if there is a code style violation
+        enforceCheck false
+        googleJavaFormat()
+        licenseHeaderFile "${projectDir}/config/codestyle/HEADER.txt"
+        //lets you specify code that you don't want to violate rules or be reformatted
+        toggleOffOn()
+    }
+}
```

Comment:
@lindseyburnett few tiny changes I'd like to suggest:
1. move this to the `allprojects` section; you'll need to manually apply the plugin there too.
2. drop `target`, we can use the built-in source set detection
3. use `${rootDir}` instead of `${projectDir}` since we keep some projects in subdirs.

I applied this locally and was able to repro your spotlessApply results 1:1. Snippet below:

```gradle
    apply plugin: "com.diffplug.spotless"
    spotless {
        java {
            enforceCheck false
            googleJavaFormat()
            licenseHeaderFile "${rootDir}/config/codestyle/HEADER.txt"
            toggleOffOn()
        }
    }
```

There are a couple extra benefits to this too; namely:
1. Can run `spotlessCheck`/`spotlessApply` on the subprojects alone if wanted (e.g. `./gradlew rbac-client:spotlessCheck`)
2. No need to manually exclude files (e.g. I have `.idea/fileTemplates/internal/Class.java` locally that I'd need to exclude with the previous way).

Reply:
spotlessApply worked as expected, but the check step of it is checking generated code in the `build` directory

`./gradlew clean build spotlessCheck`

```
The following files had format violations:
build/generated-sources/js2p/org/candlepin/subscriptions/json/Event.java
...
Violations also present in 21 other files.
```